nexus_constructor/component_fields.py

```diff
@@ -163,8 +161,10 @@ def name(self):
     def dtype(self):
         if self.field_type == FieldType.scalar_dataset:
             return self.value.dtype
-        if self.field_type == FieldType.array_dataset:
+        elif self.field_type == FieldType.array_dataset:
```

Comment:
I think it is generally advised to stick with `if` statements in the case that there is a `return` in the previous statement.